sdk/src/suins-transaction.ts

```diff
@@ -0,0 +1,317 @@
+				this.transactionBlock.splitCoins(this.transactionBlock.gas, [
+					this.transactionBlock.pure.u64(price),
+				]),
+				this.transactionBlock.object(SUI_CLOCK_OBJECT_ID),
+			],
+		});
+
+		return nft;
+	}
+
+	createSubName({
+		parentNft,
+		name,
+		expirationTimestampMs,
+		allowChildCreation,
+		allowTimeExtension,
+	}: {
+		parentNft: ObjectArgument;
+		name: string;
+		expirationTimestampMs: number;
+		allowChildCreation: boolean;
+		allowTimeExtension: boolean;
+	}) {
+		if (!isValidSuiNSName(name)) throw new Error('Invalid SuiNS name');
+		const isParentSubdomain = isNestedSubName(name);
+		if (!this.#suinsClient.constants.suinsObjectId) throw new Error('SuiNS Object ID not found');
+		if (!this.#suinsClient.constants.subNamesPackageId)
+			throw new Error('Subdomains package ID not found');
+		if (isParentSubdomain && !this.#suinsClient.constants.tempSubNamesProxyPackageId)
+			throw new Error('Subdomains proxy package ID not found');
+
+		const subNft = this.transactionBlock.moveCall({
+			target: isParentSubdomain
+				? `${this.#suinsClient.constants.tempSubNamesProxyPackageId}::subdomain_proxy::new`
+				: `${this.#suinsClient.constants.subNamesPackageId}::subdomains::new`,
+			arguments: [
+				this.transactionBlock.object(this.#suinsClient.constants.suinsObjectId),
+				this.transactionBlock.object(parentNft),
+				this.transactionBlock.object(SUI_CLOCK_OBJECT_ID),
+				this.transactionBlock.pure.string(name),
+				this.transactionBlock.pure.u64(expirationTimestampMs),
+				this.transactionBlock.pure.bool(!!allowChildCreation),
+				this.transactionBlock.pure.bool(!!allowTimeExtension),
+			],
+		});
+
+		return subNft;
+	}
+
+	/**
+	 * Builds the PTB to create a leaf subdomain.
+	 * Parent can be a `SuinsRegistration` or a `SubDomainRegistration` object.
+	 * Can be passed in as an ID or a TransactionArgument.
+	 */
+	createLeafSubName({
+		parentNft,
+		name,
+		targetAddress,
+	}: {
+		parentNft: ObjectArgument;
+		name: string;
+		targetAddress: string;
+	}) {
+		if (!isValidSuiNSName(name)) throw new Error('Invalid SuiNS name');
+		const isParentSubdomain = isNestedSubName(name);
+		if (!this.#suinsClient.constants.suinsObjectId) throw new Error('SuiNS Object ID not found');
+		if (!this.#suinsClient.constants.subNamesPackageId)
+			throw new Error('Subdomains package ID not found');
+		if (isParentSubdomain && !this.#suinsClient.constants.tempSubNamesProxyPackageId)
+			throw new Error('Subdomains proxy package ID not found');
+
+		this.transactionBlock.moveCall({
+			target: isParentSubdomain
+				? `${this.#suinsClient.constants.tempSubNamesProxyPackageId}::subdomain_proxy::new_leaf`
+				: `${this.#suinsClient.constants.subNamesPackageId}::subdomains::new_leaf`,
+			arguments: [
+				this.transactionBlock.object(this.#suinsClient.constants.suinsObjectId),
+				this.transactionBlock.object(parentNft),
+				this.transactionBlock.object(SUI_CLOCK_OBJECT_ID),
+				this.transactionBlock.pure(name),
```

Comment:
this form of pure has been deprecated for a while, does typescript not complain about this?

Reply:
Interestingly, it doesn't, but good eyes 😄